Dockerfile

```diff
@@ -11,8 +11,8 @@ COPY apis/go.mod apis/go.mod
 COPY apis/go.sum apis/go.sum
 COPY hack/tools/go.mod hack/tools/go.mod
 COPY hack/tools/go.sum hack/tools/go.sum
-COPY pkg/ironic/go.mod pkg/ironic/go.mod
-COPY pkg/ironic/go.sum pkg/ironic/go.sum
+COPY pkg/bmc/go.mod pkg/bmc/go.mod
```

Comment:
Is it worth putting the bmc package under something like `pkg/<something>/bmc` so that we can easily add more packages to the module if that makes sense in the future?

I would suggest 'hardware', but there is already a package with that name :(

Reply:
`bmc` package is highly dependent on ironic implementation itself, it does not do real BMC communication itself, so putting it to `pkg/bmc` could be somewhat confusing

Reply:
What do you think about pkg/hardwareutils/bmc? 

Hardwareutils will be the module.

Reply:
That WFM.

Reply:
Done.

Reply:
-1 to hardwareutils. what's the point of it? we can add more packages if we need it, the initial argument doesn't seem correct to me, unless we want to publish these packages to some global registry (in which case hardwareutils is also a terrible name).

Reply:
What if we use `ironic` as module name. But currently it only includes bmc package for now. 

Thanks to that, we will have ability to move other ironic packages in this module time after time (after decoupling apis and provisioner dependency). That looks like a good path to create encapsulated ironic module as we intended.

Reply:
> -1 to hardwareutils. what's the point of it?

Well, the point is that we need an extra layer in the directory hierarchy and the name `hardware` is already taken.

> we can add more packages if we need it,

Note that we can't if the bmc package remains under the top level `/pkg/` dir and not one level deeper. Or we could, but they'd all appear under `pkg/bmc/...`.

> the initial argument doesn't seem correct to me, unless we want to publish these packages to some global registry (in which case hardwareutils is also a terrible name).

To be clear, the 'name' would be `github.com/metal3-io/baremetal-operator/pkg/hardwareutils` (there is no global registry as such though, although there is a global module cache).

Reply:
Wait, so a directory does not become a package, only a parent directory? So if we make `pkg/bmc` a module it owns everything in `pkg`? O__o Please tell me you're kidding.

Reply:
No, but if we put go.mod in `pkg/bmc` then that is the module and any new packages in that module will appear like `pkg/bmc/somethingusedbyapibutnotbmcrelated`. So putting go.mod in `pkg/hardwareutils` and the bmc package in `pkg/hardwareutils/bmc` is more flexible because we can add `pkg/hardwareutils/somethingusedbyapibutnotbmcrelated` later.

Reply:
I don't think I see the problem in creating a new module for `somethingusedbyapibutnotbmcrelated`, but I guess we don't want to end up in an infinite bike-shedding either.

Reply:
Agree. I don't want neither. Because interestingly this PR becomes bottleneck for independent different PRs.

Reply:
Thanks @dtantsur

Reply:
> I don't think I see the problem in creating a new module

New modules are a pain. We have to handle them in every Makefile target, and maintain replacement directives for all of them (with matching versions) in every downstream repo. The fewer we have, the better.